### PR TITLE
Julia: correct bugs with struct constructor

### PR DESCRIPTION
**Pull request by getzze**
Correct three bugs with functions and constructors:
- `lenght(a) == 0` was recognized as a short function definition.
- Parametric constructors with `function` keyword were not recognized, like `function MyStruct{Int}(x) ... end`
- Inner constructors inside `struct` defined below a constructor with the `function` keyword had incorrect scope, it was considered to be below the constructor with `function`.
- [EDIT] Outer constructor, like parametric short function `func{Int}(x) = 0`, was not recognized as a short function.

### CODE REVIEW

File: parsers/julia.c [hunk 1]
```diff
@@ -1072,16 +1072,28 @@ static void parseFunction (lexerState *lexer, vString *scope, int parent_kind)
...
+    else if (lexer->cur_c == '.')
+    {
         local_scope = vStringNewCopy(lexer->token_str);
         local_parent_kind = K_MODULE;
         advanceChar(lexer);
         advanceToken(lexer, true, false);
-    } else {
+    }
+    else
+    {
```
Comment (masatake) [inline parsers/julia.c:1084]: I'm sorry that I accepted this style of change. I should not.

This is nothing to do with the original issue you wanted to fix.
Could you make a separate commit for the change of style?
In that case, I would like to use "Julia,cosmetic" or "Julia,style" as the prefix for the header of the commit log.

INSTEAD, I would like you to invite this project. So you can control your parser especially when we get a pull request for your parser. We can avoid the mistake I took.

Comment (masatake) [reply parsers/julia.c:1084]: You can find "cosmetic" change with searching "cosmetic in git log output.

**Comment by getzze** [comment]
Now I think it's working, but the code for function and short-function should be refactored to avoid these kinds of bugs, I will add it to the TODO list.

**Review by masatake — APPROVED** [review]

**Comment by masatake** [comment]
Please, use "Create a merge commit".

**Pull request merged**